### PR TITLE
add x0, sx0 and parameter field to ExpData

### PR DESCRIPTION
**Pull request by FFroehlich**
These changes will allow us to replace parts of the loop in `pypesto.AmicyObjective` with `runAmiciSimulations`, which in turns allows multithreading in pypesto objectives.

### CODE REVIEW

**Comment by FFroehlich** [comment]
WIP: test this new feature

File: include/amici/edata.h [hunk 1]
```diff
@@ -10,51 +10,51 @@ namespace amici {
...
+ * condition-specific data */
 class ExpData {
 
   public:
-    /** default constructor */
+    /** @brief default constructor */
     ExpData();
 
-    /** default copy constructor, needs to be declared to be generated in swig*/
+    /** @brief default copy constructor, needs to be declared to be generated in
```
Comment (dweindl) [inline include/amici/edata.h]:
```suggestion
    /** @brief Copy constructor, needs to be declared to be generated in
```

File: include/amici/edata.h [hunk 2]
```diff
@@ -65,22 +65,21 @@ class ExpData {
...
-     * constructor that initializes with Model
+     * @brief constructor that initializes with Model
      *
      * @param model pointer to model specification object
      */
     explicit ExpData(const Model &model);
 
     /**
-     * constructor that initializes with returnData, adds
+     * @brief constructor that initializes with returnData, adds
```
Comment (dweindl) [inline include/amici/edata.h]: something missing?

File: include/amici/edata.h [hunk 3]
```diff
@@ -89,84 +88,87 @@ class ExpData {
     ExpData(const ReturnData &rdata, realtype sigma_y, realtype sigma_z);
 
     /**
-     * constructor that initializes with returnData, adds
+     * @brief constructor that initializes with returnData, adds
```
Comment (dweindl) [inline include/amici/edata.h]: "adds"?

**Comment by FFroehlich** [comment]
> The urge to rename (or split) `ExpData` is getting stronger...

I know 😅

**Pull request merged**